### PR TITLE
Resolve issues for IntelliJ toolkit

### PR DESCRIPTION
**Pull request by Flanker32**
What does this implement/fix? Explain your changes.
---------------------------------------------------
- Filter out existing resource groups from draft items in ResourceGroupComboBox, [AB#1982621](https://dev.azure.com/mseng/a4d27ce2-a42d-4b71-8eef-78cee9a9728e/_workitems/edit/1982621)
- Add action to open resource connection explorer, [AB#1979981](https://dev.azure.com/mseng/a4d27ce2-a42d-4b71-8eef-78cee9a9728e/_workitems/edit/1979981)
- Add validation for parent module in function creation wizard, [AB#1976748](https://dev.azure.com/mseng/a4d27ce2-a42d-4b71-8eef-78cee9a9728e/_workitems/edit/1976748)
- Modify subscription table model in dispatch thread, [AB#1973165](https://dev.azure.com/mseng/a4d27ce2-a42d-4b71-8eef-78cee9a9728e/_workitems/edit/1973165)

Does this close any currently open issues?
------------------------------------------

Any relevant logs, screenshots, error output, etc.?
-------------------------------------

Any other comments?
-------------------
…

Has this been tested?
---------------------------
- [ ] Tested

### CODE REVIEW

File: PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-resource-connector-lib/src/main/resources/META-INF/azure-intellij-resource-connector-lib.xml [hunk 1]
```diff
@@ -38,5 +38,7 @@
             text="Connect to Azure Resource"
             icon="/icons/connector/connect.svg"
             description="To consume azure resources in your project"/>
+        <action class="com.microsoft.azure.toolkit.intellij.connector.actions.OpenResourceConnectionExplorerAction" id="AzureToolkit.OpenResourceConnectionExplorerAction"
+                text="Open Azure Resource Connection Explorer" description="Open Azure Resource Connection Explorer" icon="/icons/Common/AzureResourceConnector.svg"/>
```
Comment (Flanker32) [inline PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-resource-connector-lib/src/main/resources/META-INF/azure-intellij-resource-connector-lib.xml]: -> Connections

File: PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-guidance/src/main/java/com/microsoft/azure/toolkit/ide/guidance/Context.java [hunk 2]
```diff
@@ -76,8 +76,9 @@ public String render(final String template) {
             final Map<String, Object> bindings = new HashMap<>();
             bindings.put("context", this.parameters);
             final Template tpl = engine.createTemplate(template);
-            // for not exists values, engine will render them as null, remove them in the final result
-            return tpl.make(bindings).toString().replace("null", "");
+            return tpl.make(bindings).toString()
+                    .replace("null", "") // for not exists values, engine will render them as null, remove them in the final result
+                    .replace("  ", " "); // remove extra spaces
```
Comment (Flanker32) [inline PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-guidance/src/main/java/com/microsoft/azure/toolkit/ide/guidance/Context.java]: handle cases with multiple space

File: PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-resource-connector-lib/src/main/java/com/microsoft/azure/toolkit/intellij/connector/AzureServiceResource.java [hunk 3]
```diff
@@ -56,8 +56,8 @@ public synchronized T getData() {
     @Override
     public Map<String, String> initEnv(Project project) {
         final T resource = this.getData();
-        if (!resource.exists()) {
-            throw new AzureToolkitRuntimeException(String.format("'%s' doesn't exist.", resource));
+        if (resource == null || !resource.exists()) {
+            throw new AzureToolkitRuntimeException(String.format("%s '%s' does not exist.", this.id.resourceType(), this.getName()));
```
Comment (Flanker32) [inline PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-resource-connector-lib/src/main/java/com/microsoft/azure/toolkit/intellij/connector/AzureServiceResource.java]: may change to resourceTypeName later

File: PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-resource-connector-lib/src/main/java/com/microsoft/azure/toolkit/intellij/connector/AzureServiceResource.java [hunk 4]
```diff
@@ -73,6 +77,19 @@ public String getName() {
         return this.id.name();
     }
 
+    public String getResourceType() {
+        final ResourceId parent = this.id.parent();
```
Comment (Flanker32) [inline PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-resource-connector-lib/src/main/java/com/microsoft/azure/toolkit/intellij/connector/AzureServiceResource.java:81]: change to `getOrTemp(id).getResourceTypeName()`

File: PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-lib/src/main/java/com/microsoft/azure/toolkit/intellij/common/component/resourcegroup/ResourceGroupNameTextField.java [hunk 5]
```diff
@@ -30,7 +30,9 @@ public ResourceGroupNameTextField() {
 
     public void setSubscription(Subscription subscription) {
         this.subscription = subscription;
-        this.validateValueAsync();
+        if (this.getValidationInfo() != null) {
```
Comment (Flanker32) [inline PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-lib/src/main/java/com/microsoft/azure/toolkit/intellij/common/component/resourcegroup/ResourceGroupNameTextField.java:35]: add comment here

**Review by wangmingliang-ms — APPROVED** [review]
LGTM

**Pull request merged**